src/common_lib/src/validator/assertions/inter_cones_distance.cpp

```diff
@@ -0,0 +1,36 @@
+#include "common_lib/validator/assertions/inter_cones_distance.hpp"
+#include <cmath>
+#include <limits>
+#include <algorithm>
+
+namespace common_lib::validator::assertions {
+
+bool InterConesDistance::validate(const custom_interfaces::msg::ConeArray& msg) const {
+    if (msg.cone_array.empty()) {
+        return true;
+    }
+    
+    auto sortedCones = msg.cone_array;
+    std::sort(sortedCones.begin(), sortedCones.end(), [](const auto& a, const auto& b) {
+        return (a.position.x < b.position.x) || (a.position.x == b.position.x && a.position.y < b.position.y);
+    });
+
+    double min_distance = std::numeric_limits<double>::infinity();
+
+    for (size_t i = 1; i < sortedCones.size(); ++i) {
```

Comment:
This seems like it will raise problems, as the sorting is primarily done by x, which does not ensure the closest cones